include/envelope.h

```diff
@@ -66,42 +66,47 @@ class Envelope {
 public:
 	Envelope(const char* name);
 
-	void Process(const bool is_stereo, AudioFrame &frame);
+	void Process(bool is_stereo, AudioFrame& frame);
```

Comment:
I've recently learned this, and it makes sense to me:

For primitive types, the `const`ness of a function argument is an implementation detail. As this is not a pointer or a reference to some data that can be modified by the caller, from the point of view of the interface, it does not matter whether the implementation allows the primitive type to be changed inside the function or not. So it unnecessarily clutters the interface; it's not a significant detail.

For pointers and references of course it is _very_ significant whether the data the point to can be modified by the function or not; I don't even think the compiler will allow you to drop the const-ness from the declaration because it's significant.

See these more detailed discussions here:

https://stackoverflow.com/a/706071
https://stackoverflow.com/a/15297695

Reply:
I agree that const in the case of by-value passing of parameters is only significant for the implementation, so language-wise there's a point.

However, consistency between definition and implementation is also important. A well aimed find (and replace) for example becomes a tad more difficult if const is in one place, but not in the other. Although it is not important to the caller, is might be important to the developer.

Reply:
Ok, had a brief chat with the guys, and it's the least surprising to just keep them in sync. It's a very minor thing and arguably it does not really matter either way, so let's just prefer consistency then like you said.